synapse/storage/background_updates.py

```diff
@@ -158,11 +168,16 @@ def __init__(self, hs: "HomeServer", database: "DatabasePool"):
 
         self._background_update_performance: Dict[str, BackgroundUpdatePerformance] = {}
         self._background_update_handlers: Dict[str, _BackgroundUpdateHandler] = {}
+        # TODO: all these bool flags make me feel icky---can we combine into a status
+        # enum?
         self._all_done = False
 
         # Whether we're currently running updates
         self._running = False
 
+        # Marker to be set if we abort and halt all background updates.
+        self._aborted = False
```

Comment:
This var seems to never be written for now, is it a provision for future use ?

Reply:
OOOOOPS.